submissions/api.py

```diff
@@ -684,6 +685,12 @@ def reset_score(student_id, course_id, item_id):
             item_id=item_id,
         )
 
+        if clear_state:
+            # sever the link between this student item and any submissions it may current have
+            for sub in student_item.submission_set.all():
+                sub.student_item = None
+                sub.save(update_fields=['student_item'])
```

Comment:
@mulby We're a little curious about how this will impact analytics.

Essentially what happens on a "reset state" request is that we leave the `Submission` item in the database, and when the `StudentModule` is deleted (https://github.com/edx/edx-platform/blob/master/lms/djangoapps/instructor/enrollment.py#L246-L253), the only possible reference to the submission (via its uuid) is lost.

However, with the addition of staff tools, now a third party (the staff user) can construct a triplet of (user/course/problem location) and use *that* as a key to access a supposedly orphaned submission. The way we're attempting to address the issue in this PR is to null out the key for an orphaned submission, which will then be persisted in the database.

Will this be an issue for analytics? It will make it possible for `Submission` data to exist that cannot be easily linked to a student. Would adding events coverage make it any better?